### DISCUSSION TITLE
Emit warning if trying to add a file ignored via `.chezmoiignore`

### DISCUSSION THREAD

**Discussion by Blacksmoke16** [twpayne/chezmoi#2753] [Ideas]
I tried to `chezmoi add` a file today but couldn't get it to work. Command would exit successfully, but just wouldn't add the file. After a while I figured out the file I wanted to add was actually in a directory thats ignored via `.chezmoiignore` that I forgot about.

Might be a nice QoL change to emit a warning or something in this case to make it easier to know what the problem is.

**Comment by Lockszmith-GH** [comment]
did you try `--debug`?

**Reply by Blacksmoke16** [reply]
Yes, but nothing really jumped out at me as being obvious. Given there was like 40 lines of output and a few seemingly unrelated errors. Something like this would be a lot better:

```shell
$ chezmoi add path/to/thing.txt
chezmoi: Not adding 'path/to/thing.txt' since it is ignored via '.chezmoiignore'
```

**Reply by Lockszmith-GH** [reply]
It wasn't a criticism, I was asking if `--debug` already caught it.

I'm guessing the reason it's suppressed by default is to prevent cli-globing from generating errors.

In Linux/Unix when you do a `chezmoi add *` - the * is translated by the shell, not by chezmoi.
So that would be noisy.

**Comment by twpayne** [comment]
I agree that this is a nice QoL improvement. #2762 implements it.
